routes/admin_tutorial.go

```diff
@@ -0,0 +1,213 @@
+		userMetadata.IsFeaturedTutorialUndiscoveredCreator = false
+	} else {
+		userMetadata.IsFeaturedTutorialUndiscoveredCreator = !requestData.IsRemoval
+		userMetadata.IsFeaturedTutorialWellKnownCreator = false
+	}
+
+	if err = fes.putUserMetadataInGlobalState(userMetadata); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminResetJumioForPublicKey: Problem putting updated user metadata in Global state: %v", err))
+		return
+	}
+}
+
+func (fes *APIServer) AdminGetTutorialCreators(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := AdminGetTutorialCreatorsRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminUpdateTutorialCreator: Problem parsing request body: %v", err))
+		return
+	}
+	wellKnownSeekKey := _GlobalStateKeyWellKnownTutorialCreators
+	undiscoveredSeekKey := _GlobalStateKeyUndiscoveredTutorialCreators
+	maxKeyLen := 1 + btcec.PubKeyBytesLenCompressed
+
+	wellKnownKeys, _, err :=fes.GlobalStateSeek(
+		wellKnownSeekKey,
+		wellKnownSeekKey,
+		maxKeyLen,
+		300,
+		false,
+		false,
+	)
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"AdminGetTutorialCreators: Problem seeking through global state keys: #{err}"))
+		return
+	}
+
+	undiscoveredKeys, _, err := fes.GlobalStateSeek(
+		undiscoveredSeekKey,
+		undiscoveredSeekKey,
+		maxKeyLen,
+		300,
+		true,
+		false,
+	)
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"AdminGetTutorialCreators: Problem seeking through global state keys: #{err}"))
+		return
+	}
+	var undiscoveredLimit int
+	if len(undiscoveredKeys) < requestData.ResponseLimit {
+		undiscoveredLimit = len(undiscoveredKeys)
+	} else {
+		undiscoveredLimit = requestData.ResponseLimit
+	}
+
+	var undiscoveredPublicKeysBase58Check []string
+	ShuffleKeys(&undiscoveredKeys)
+	for _, dbKeyBytes := range undiscoveredKeys[:undiscoveredLimit] {
+		// Chop the public key out of the db key.
+		// The dbKeyBytes are: [One Prefix Byte][btcec.PubKeyBytesLenCompressed]
+		publicKeyBytes := dbKeyBytes[1 :][:]
+		publicKeyBase58Check := lib.Base58CheckEncode(publicKeyBytes, false, fes.Params)
+		undiscoveredPublicKeysBase58Check = append(undiscoveredPublicKeysBase58Check, publicKeyBase58Check)
+	}
+
+	var wellKnownLimit int
+	if len(wellKnownKeys) < requestData.ResponseLimit {
+		wellKnownLimit = len(wellKnownKeys)
+	} else {
+		wellKnownLimit = requestData.ResponseLimit
+	}
+	var wellKnownPublicKeysBase58Check []string
+	ShuffleKeys(&wellKnownKeys)
+	for _, dbKeyBytes := range wellKnownKeys[:wellKnownLimit] {
+		// Chop the public key out of the db key.
+		// The dbKeyBytes are: [One Prefix Byte][btcec.PubKeyBytesLenCompressed]
+		publicKeyBytes := dbKeyBytes[1 :][:]
+		publicKeyBase58Check := lib.Base58CheckEncode(publicKeyBytes, false, fes.Params)
+		wellKnownPublicKeysBase58Check = append(wellKnownPublicKeysBase58Check, publicKeyBase58Check)
+	}
```

Comment:
Seems like a chunk of this function could be refactored. We have duplicate code for "well known" and "undiscovered". Let's create a helper function